pkg/scheduler/actions/backfill/backfill.go

```diff
@@ -68,6 +74,101 @@ func (alloc *backfillAction) Execute(ssn *framework.Session) {
 			}
 		}
 	}
+
+	if len(alloc.arguments) == 0 {
+		return
+	}
+
+	backfillNonBestEffortJobsEnabled, err := strconv.ParseBool(alloc.arguments[conf.BackfillFlagName])
+	if err != nil {
+		glog.V(3).Infof("backfilling non best effort jobs is disabled")
+		return
+	}
+
+	if backfillNonBestEffortJobsEnabled {
+		// Collect back fill candidates
+		backFillCandidates := make([]*api.JobInfo, 0, len(ssn.Jobs))
+		for _, job := range ssn.Jobs {
+			if !ssn.BackFillEligible(job) {
+				continue
+			}
+			backFillCandidates = append(backFillCandidates, job)
+		}
+
+		// Release resources allocated to unready top dog jobs so that
+		// we can back fill more jobs in the next step.
+		resourceReleased := false
+		for _, job := range ssn.Jobs {
+			if ssn.JobReady(job) || ssn.JobBackfillReady(job) {
+				glog.V(3).Infof("Disable backfill on job <%v/%v>", job.Namespace, job.Name)
+			} else {
+				releaseReservedResources(ssn, job)
+				resourceReleased = true
+			}
+		}
+
+		if resourceReleased {
+			for _, job := range backFillCandidates {
+				backFill(ssn, job)
+			}
+		}
+	}
+}
+
+// Releases resources allocated to the given job back to the cluster.
+func releaseReservedResources(ssn *framework.Session, job *api.JobInfo) {
```

Comment:
Prefer to add this func to session as `Deallocate`.

Reply:
no. this function needs to be called before backfill()

Reply:
move this func as a member of session, and renamed it to `Deallocate`. So we can reuse it later.